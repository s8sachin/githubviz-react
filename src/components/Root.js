import React, { Component } from 'react';
import { Grid, Jumbotron, Button } from 'react-bootstrap';
import Header from './commons/Header';

class Root extends Component {

  clickFunc() {
    console.log("hi")
  }
  render() {
    return (
      <div>
        <Header/>
        <Grid bsClass='container'>
          <Jumbotron>
            <h1>Hello, world!</h1>
            <p>
              This is a simple hero unit, a simple jumbotron-style component for calling
              extra attention to featured content or information.
            </p>
            <p>
              <Button bsStyle="primary" href='#'>Learn more</Button>
            </p>
          </Jumbotron>
        </Grid>
      </div>
    );
  }
}

export default Root;