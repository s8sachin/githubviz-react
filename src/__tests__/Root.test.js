import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import Root from '../components/Root';
import { shallow, mount, render, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import LocalStorageMock from '../LocalStorageMock';

global.localStorage = new LocalStorageMock;

configure({ adapter: new Adapter() });

describe('Root', () => {
  it('should mount Root component', () => {
    const wrapper = shallow(<Root />)
    expect(toJson(wrapper)).toMatchSnapshot();
  })

  // it("should click create blog post link", () => {
  //   const wrapper = shallow(<Home />)
  //   const card = wrapper.find('Columns').find('Box').first().find('Card')
  //   const component = renderer.create(card.props().link)
  //   let tree = component.toJSON();
  //   tree.props.onClick();
  //   expect(tree).toMatchSnapshot();
  // })

  // it("should click sample table link", () => {
  //   const wrapper = shallow(<Home />)
  //   const card = wrapper.find('Columns').find('Box').at(2).find('Card')
  //   const component = renderer.create(card.props().link)
  //   let tree = component.toJSON();
  //   tree.props.onClick();
  //   expect(tree).toMatchSnapshot();
  // })

  // it("should click posts feed link", () => {
  //   const wrapper = shallow(<Home />)
  //   const card = wrapper.find('Columns').find('Box').at(1).find('Card')
  //   const component = renderer.create(card.props().link)
  //   let tree = component.toJSON();
  //   tree.props.onClick();
  //   expect(tree).toMatchSnapshot();
  // })
})