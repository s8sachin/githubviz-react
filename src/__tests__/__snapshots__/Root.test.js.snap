// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root should mount Root component 1`] = `
<div>
  <Header />
  <Grid
    bsClass="container"
    componentClass="div"
    fluid={false}
  >
    <Jumbotron
      bsClass="jumbotron"
      componentClass="div"
    >
      <h1>
        Hello, world!
      </h1>
      <p>
        This is a simple hero unit, a simple jumbotron-style component for calling extra attention to featured content or information.
      </p>
      <p>
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="primary"
          disabled={false}
          onClick={[Function]}
        >
          Login
        </Button>
      </p>
    </Jumbotron>
  </Grid>
</div>
`;
